Tests/webcomponents/js/ez-universaldiscovery.js

```diff
@@ -0,0 +1,339 @@
+            });
+        });
+    });
+
+    describe('UDW init', function () {
+        let renderViewStub;
+        const container = document.createElement('div');
+        const location = getModelMock();
+        const content = getModelMock();
+        const contentType = getModelMock();
+
+        function assertEventParam(e) {
+            assert.equal(e.detail.selection.content, modelObject);
+            assert.equal(e.detail.selection.location, modelObject);
+            assert.equal(e.detail.selection.contentType, modelObject);
+            assert.isFalse(e.bubbles);
+        }
+
+        beforeEach(function () {
+            universalDiscovery = fixture('BasicTestFixture');
+            defineGlobals();
+        });
+
+        afterEach(function () {
+            delete window.eZ;
+            renderViewStub.restore();
+        });
+
+        it('should append the YUI view container', function () {
+            const view = getViewMock(container);
+
+            renderViewStub = sinon.stub(window.eZ.YUI.app, 'renderSideView', function (View, Service, params, done) {
+                assert.equal(
+                    universalDiscovery.getAttribute('title'),
+                    params.title,
+                    "The title should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.multiple,
+                    params.multiple,
+                    "The multiple should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('visible-method'),
+                    params.visibleMethod,
+                    "The visibleMethod should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('confirm-label'),
+                    params.confirmLabel,
+                    "The confirmLabel should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('starting-location-id'),
+                    params.startingLocationId,
+                    "The startingLocationId should be passed in the parameters"
+                );
+
+                assert.equal(
+                    universalDiscovery.getAttribute('min-discover-depth'),
+                    params.minDiscoverDepth,
+                    "The minDiscoverDepth should be passed in the parameters"
+                );
+
+                done(false, 'whatever', view);
+            });
+            document.dispatchEvent(new CustomEvent('ez:yui-app:ready'));
+            assert.equal(
+                container.parentNode,
+                universalDiscovery
+            );
+        });
+
+        it('should have a cancelDiscoverHandler', function () {
+            universalDiscovery.addEventListener('ez:cancel', function (e) {
+                assert.isFalse(e.bubbles);
```

Comment:
Note: that's incomplete. You only check that the event is not bubbling but you never check that the event is actually dispatched, if for some reason it's not the test won't fail. In any case, that will be moved to `events > ez:cancel > should be dispatched` but that also needs to be improved.